iroha/src/kura.rs

```diff
@@ -180,77 +180,129 @@ impl Default for Mode {
 #[derive(Debug)]
 pub struct BlockStore {
     path: PathBuf,
+    blocks_per_file: u64,
 }
 
 impl BlockStore {
     /// Initialize block storage at `path`.
     ///
     /// # Errors
     /// - Failed to create directory.
-    pub fn new(path: &Path) -> Result<BlockStore> {
+    pub fn new(path: &Path, blocks_per_file: u64) -> Result<BlockStore> {
         if fs::read_dir(path).is_err() {
             fs::create_dir_all(path).wrap_err("Failed to create Block Store directory.")?;
         }
         Ok(BlockStore {
             path: path.to_path_buf(),
+            blocks_per_file: if blocks_per_file == 0 { config::DEFAULT_BLOCKS_PER_STORAGE_FILE } else { blocks_per_file },
         })
     }
 
-    fn get_block_filename(block_height: u64) -> String {
-        format!("{}", block_height)
+    /// Returns expected filename of datafile containing block, provided its height
+    async fn get_block_filename(dir: &PathBuf, block_height: u64, blocks_per_file: u64) -> Result<String> {
+        let storage_indices = sorted_storage_files_base_indices(dir).await
+            .wrap_err("Failed to list storage files")?;
+        match storage_indices.last() {
+            Some(max_base) => {
+                assert!(block_height > *max_base);
```

Comment:
```suggestion
                debug_assert!(block_height > *max_base);
```

Reply:
As `assert!` will be in release build also, while debug will be compiled out

Reply:
So, should I use Result as a return type? Or just truncate successfull deseralization streak and log error? Which is more appropriate?

Reply:
Actually yes better to use Result than asserts.
Create `BlockStoreError` enum with variants like `InconsistentHeight(...)`, `FailedToRead` ...
And then return Result<_, BlockStoreError>

Reply:
I think your point about not truncating but failing for ease of administration - is valid

Reply:
Then I will adhere to this logic:
1. Managed to deserialized - success
2. Anything went wrong - report an error